OrbitSsh/Session.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitSsh/Session.h"
+
+#include <libssh2.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitSsh/ResultType.h"
+#include "OrbitSsh/Socket.h"
+
+namespace OrbitSsh {
+
+Session::~Session() { libssh2_session_free(raw_session_ptr_); }
+
+ResultType Session::Initialize() {
```

Comment:
Make the constructor private and make this a factory function.

Reply:
I am making a factory function that returns `std::unique_ptr<Session>`. 

There is a tradeoff to be made, either make the constructor public and use `std::make_unique` or make the constructor private and use `std::unique_ptr(new Session(...))`. 
What do you think?

Reply:
Why an allocation? It's not an inherited type. Just return a outcome::result<Session> or a std::optional<Session>

Reply:
done